### DISCUSSION TITLE
Private file uploads with random name, but show original name in UP

### DISCUSSION THREAD

**Discussion by LeeMangold** [filamentphp/filament#14182] [Help]
### Package

Form builder

### Package Version

v3.2

### How can we help you?

[This was an issue in 2022 ](https://github.com/filamentphp/filament/discussions/3996)that and I'm wondering if there is any fix for it?

I need a user to be able to upload a private file, save, then go back and edit the model they saved the file on and add/remove multiple files to that model. Currently, if they upload test.doc, they come back to see [random string].doc instead. If this isn't possible, does anyone have a recommendation?

**Comment by leandrocfe** [comment]
Are you using `->preserveFilenames()` ?

**Reply by LeeMangold** [reply]
No - I'm concerned about the big warning about Remote Code Execution from the docs. It there something I can do to avoid that issue? (note: Creating a dynamic folder would suffice, but I'm still concerned about the vuln.)

> Are you using `->preserveFilenames()` ?
